### PR TITLE
docparams extension considers type comments as type documentation.

### LINKED ISSUE #6287: docparams should support missing type documentation when type comments are used
### Current problem

When running pylint (`pylint --disable=all --enable=missing-type-doc --load-plugins=pylint.extensions.docparams code.py`) with the following code, no errors are displayed:

```python
def variant(node: str) -> list[str]:
    """Get the variants for the given `node`.

    Args:
        node: The node to get the variants for.

    Returns:
        The available variants.

    """
    return ["hi"]
```

However when type comments are used, a `missing-type-doc` warning is shown:

```python
def variant(node):
    # type: (str) -> list[str]
    """Get the variants for the given `node`.

    Args:
        node: The node to get the variants for.

    Returns:
        The available variants.

    """
    return ["hi"]
```

### Desired solution

Ideally docparams would allow type comments to count as type documentation, just like regular type annotations are allowed.
This will require changes on these lines: https://github.com/PyCQA/pylint/blob/a6cb836da56b051cdc0d584d726fb7579415cb54/pylint/extensions/docparams.py#L560-L563

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AWhetter**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|     | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
|     | :hammer: Refactoring   |
|     | :scroll: Docs          |

## Description

The docparams extension won't raise a `missing-type-doc` when type annotations are given. This changes means that it also won't raise a  `missing-type-doc` when type comments are given.

Closes #6287

### CODE REVIEW

File: pylint/extensions/_check_docs_utils.py
Comment (Pierre-Sassoulas) [inline pylint/extensions/_check_docs_utils.py:168]:
```suggestion
def _merge_annotations(annotations: List[nodes.?], comment_annotations: List[nodes.?]) -> Generator:
```
Not sure if we can be more specific than NodeNG here..

**Comment by Pierre-Sassoulas** [comment]
What do you think of this astroid issue https://github.com/PyCQA/astroid/issues/1508 @AWhetter ?

File: pylint/extensions/_check_docs_utils.py
Comment (AWhetter) [reply pylint/extensions/_check_docs_utils.py:168]: I don't think that we can. The astroid annotations of the attributes that we get values from to pass into this function list them only as NodeNG at the moment.

**Review by Pierre-Sassoulas — DISMISSED** [review]
LGTM, thank you 👍 I'm going to wait for another review if possible but this is definitely going in 2.16.0

File: pylint/extensions/_check_docs_utils.py
Comment (DanielNoord) [inline pylint/extensions/_check_docs_utils.py:173]: We should probably follow the style of importing from `nodes` instead of `astroid` here.

**Review by Pierre-Sassoulas — APPROVED** [review]
The CI fail on the doc is a temporary fail in reaching a twitter link.

**Review by Pierre-Sassoulas — APPROVED** [review]
Let's merge this 🎉

**Comment by Pierre-Sassoulas** [comment]
@AWhetter the pypi 3.7 tests seems genuine, but also it's on an old interpreter, should we just ignore it ?

**Comment by Pierre-Sassoulas** [comment]
I tried to add coverage ans force pushed some code but I was out of idea to get the last line @AWhetter :) we're very close to being able to merge

**Comment by AWhetter** [comment]
@Pierre-Sassoulas Thanks for adding those tests! I've included them again and added a test that covers that final line.
The test failure seems to be happening in the main branch and is unrelated to these changes. So I think that this is ready to merge.

**Review by Pierre-Sassoulas — APPROVED** [review]
Amazing, thank you for covering the last missing part 👍

**Pull request merged**
